src/ceph-volume/ceph_volume/systemd/main.py

```diff
@@ -0,0 +1,91 @@
+"""
+This file is used only by systemd units that are passing their instance suffix
+as arguments to this script so that it can parse the suffix into arguments that
+``ceph-volume <sub command>`` can consume
+"""
+import os
+import sys
+import time
+import logging
+from ceph_volume import log, process
+from ceph_volume.exceptions import SuffixParsingError
+
+
+def parse_subcommand(string):
+    subcommand = string.rsplit('-', 1)[-1]
+    if not subcommand:
+        raise SuffixParsingError('subcommand', string)
+    return subcommand
+
+
+def parse_osd_id(string):
+    osd_id = string.split('-', 1)[0]
+    if not osd_id:
+        raise SuffixParsingError('OSD id', string)
+    if osd_id.isdigit():
+        return osd_id
+    raise SuffixParsingError('OSD id', string)
+
+
+def parse_osd_uuid(string):
+    osd_id = '%s-' % parse_osd_id(string)
+    osd_subcommand = '-%s' % parse_subcommand(string)
+    # remove the id first
+    trimmed_suffix = string.split(osd_id)[-1]
+    # now remove the sub command
+    osd_uuid = trimmed_suffix.split(osd_subcommand)[0]
+    if not osd_uuid:
+        raise SuffixParsingError('OSD uuid', string)
+    return osd_uuid
+
+
+def main(args=None):
+    """
+    Main entry point for the ``ceph-volume-systemd`` executable. ``args`` are
+    optional for easier testing of arguments.
+
+    Expected input is similar to::
+
+        ['/path/to/ceph-volume-systemd', '<osd id>-<osd uuid>-<device type>']
```

Comment:
The current implementation doesn't expose anything. This executable lives in the same code base but it is a separate script for that same reason: it is not available in any of the `ceph-volume` sub-commands.

The `ceph-volume-systemd` script is not front-facing, and it is just there to accommodate for the parsing of the various other sub-systems. You are right that is kind of requiring 3 parameters, but a sub-command (e.g. nvme) could chose to add a noop. 

As it is right now, this is directly consumable by any new tech that comes around, using the same script(s) and same systemd units with no modification. All one would need to do is to hook into the internal API and call the helpers to enable the units.

If we went the route of `ceph-volume-gpt` that would mean somehow somewhere another systemd unit would need to exist. With this approach it doesn't need anything else.